work16/src/main.js

```diff
@@ -0,0 +1,298 @@
+import './style.scss';
+import { differenceInCalendarDays } from 'date-fns';
+
+const REQUEST_URL = "https://mocki.io/v1/720cde09-820e-4541-b003-c36cf2f1a414";
+
+// 間違っているURLの場合↓
+// const REQUEST_URL = "https://myjson.dit.upm.es/api/bins/ほげほげajy3";
+// 空の配列の場合↓
+// const REQUEST_URL = "https://mocki.io/v1/242b685f-a3d7-45a8-aeca-0376bd495b89";
+// 503エラーの場合↓
+// const REQUEST_URL = "https://httpstat.us/503";
+
+const init = async () => {
+  renderLoadingImg();
+  let responseJsonData;
+
+  try {
+    responseJsonData = await request();
+
+    if (!responseJsonData) {
+      return;
+    }
+  } catch (error) {
+    displayErrorMessage(error.message);
+  }
+  finally {
+    hideLoadingImg();
+  }
+
+  if (!responseJsonData.data.length) {
+    displayErrorMessage("data is empty");
+    return;
+  }
+  displayNews(responseJsonData.data);
+}
+
+const request = async () => {
+  const response = await fetch(REQUEST_URL, {
+    headers: {
+      "Content-Type": "application/json"
+    }
+  });
+  if (!response.ok) {
+    throw new Error(`${response.status}:${response.statusText}`);
+  } else {
+    return response.json();
+  }
+}
+
+const displayErrorMessage = (error) => {
+  const p = document.createElement("p");
+  p.textContent = error;
+  document.body.prepend(p);
+}
+
+const displayNews = (newsDataArray) => {
+  renderNewsTab(newsDataArray);
+  renderNewsContent(newsDataArray);
+}
+
+const renderLoadingImg = () => {
+  const div = document.createElement("div");
+  const img = document.createElement("img");
+
+  div.classList = "loading_wrap";
+  div.id = "js_loading_wrap"
+  img.src = "https://i.postimg.cc/k2RLrq7B/loading-circle.gif";
+  img.alt = "ローディング画像"
+
+  div.appendChild(img);
+  document.body.prepend(div);
+}
+
+const renderNewsTab = (newsDataArray) => {
+  const tabList = document.getElementById("js_tabList");
+  const fragment = document.createDocumentFragment();
+
+  newsDataArray.forEach((newsCategoryObj, newsCategoryIndex) => {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    let isInitialDisplay = newsCategoryObj.isInitialDisplay;
+
+    if (isInitialDisplay === "true") {
+      li.setAttribute("aria-selected", true);
+    } else {
+      li.setAttribute("aria-selected", false);
+    }
```

Comment:
IMO
It can also be written as follows
```suggestion
    li.ariaSelected = isInitialDisplay;
```

Reply:
This surprised me.
I was surprised because my code went from 4 lines to one.

commit: [4f1600b](https://github.com/yasu-pro/js-handson/pull/18/commits/4f1600b04441b169ef4fb19475a04815ea4819e4)